src/cpp/CommandLineProcessor.cpp

```diff
@@ -0,0 +1,37 @@
+#include "CommandLineProcessor.h"
+#include "CommandLineProcessorConstants.h"
+#include "TextNoteCollection.h"
+#include "FileBuilder.h"
+#include "doctest.h"
+
+void CommandLineProcessor::run(int argc, char **argv) {
+    for (int i = 0; i < argc; i++) {
+        if (argv[i] == ADD_NOTE_NAME) {
+            std::string addNoteTitle(argv[++i]);
+            addNote(addNoteTitle);
+        }
+        /*
+        else if (argv[i] == SEARCH_NOTE_NAME) {
+            // do something
+        }
+        */
+    }
+}
+
+void CommandLineProcessor::addNote(const std::string &title) {
+    // Reading a new text note from command line
+    std::string text, line;
+    std::cout << "Please, enter the text of a new note:" << std::endl;
+    while (getline(std::cin, line) && !line.empty()) {
+        text += line;
```

Comment:
This code omits newline characters from text, so multiline notes become single-line
Fixed in 82285e850ff76f9330d007d44285d179ec44d48d